src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,15 +1,276 @@
 import { Component } from 'react';
+import cn from 'classnames';
+import './NewMovie.scss';
 
-type Props = {};
-type State = {};
+type Props = {
+  addFilm: (newMovie: Movie) => void,
+};
+
+type State = {
+  newMovie: Movie,
+  titleError: boolean,
+  descriptionError: boolean,
+  imgUrlError: boolean,
+  imdbUrlError: boolean,
+  imdbIdError: boolean,
+};
 
 export class NewMovie extends Component<Props, State> {
-  state: State = {};
+  state: State = {
+    newMovie: {
+      title: '',
+      description: '',
+      imgUrl: '',
+      imdbUrl: '',
+      imdbId: '',
+    },
+    titleError: false,
+    descriptionError: false,
+    imgUrlError: false,
+    imdbUrlError: false,
+    imdbIdError: false,
+
+  };
+
+  handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const { name, value } = event.target;
+
+    this.setState((state) => ({
+      newMovie: {
+        ...state.newMovie,
+        [name]: value,
+      },
+      titleError: false,
+      descriptionError: false,
+      imdbIdError: false,
+    }));
+  };
+
+  checkFormTitle = () => {
+    if (!this.state.newMovie.title.trim()) {
+      this.setState(() => ({
+        titleError: true,
+      }));
+    }
+  };
+
+  checkFormDescription = () => {
+    if (!this.state.newMovie.description.trim()) {
+      this.setState(() => ({
+        descriptionError: true,
+      }));
+    }
+  };
+
+  validateUrl = (imgUrl: string) => {
+    // eslint-disable-next-line
+    const reg = /^((([A-Za-z]{3,9}:(?:\/\/)?)(?:[-;:&=+$,\w]+@)?[A-Za-z0-9.-]+|(?:www\.|[-;:&=+$,\w]+@)[A-Za-z0-9.-]+)((?:\/[+~%/.\w-_]*)?\??(?:[-+=&;%@.\w_]*)#?(?:[.!/\\\w]*))?)$/;
+
+    return reg.test(imgUrl);
+  };
+
+  checkFormImgUrl = () => {
+    const { imgUrl } = this.state.newMovie;
+    const res = this.validateUrl(imgUrl);
+
+    if (!res) {
+      this.setState(() => ({
+        imgUrlError: true,
+      }));
+    } else {
+      this.setState(() => ({
+        imgUrlError: false,
+      }));
+    }
+  };
+
+  checkFormImdbUrl = () => {
+    const { imdbUrl } = this.state.newMovie;
+    const res = this.validateUrl(imdbUrl);
+
+    if (!res) {
+      this.setState(() => ({
+        imdbUrlError: true,
+      }));
+    } else {
+      this.setState(() => ({
+        imdbUrlError: false,
+      }));
+    }
+  };
+
+  checkFormImdbId = () => {
+    if (!this.state.newMovie.imdbId.trim()) {
+      this.setState(() => ({
+        imdbIdError: true,
+      }));
+
+      return false;
+    }
+
+    return true;
+  };
+
+  handleSubmitForm = (e: { preventDefault: () => void; }) => {
+    e.preventDefault();
+
+    if (this.state.imdbUrlError === true) {
+      return;
+    }
+
+    if (this.state.imgUrlError === true) {
+      return;
+    }
```

Comment:
combine these blocks and destructure everything from state

```suggestion
    if (imdbUrlError || imgUrlError) {
      return;
    }
```

Reply:
fixed.